### ISSUE TITLE
FR: revset language or similar for other config values 

### ISSUE THREAD

**Issue by stacyharper** [martinvonz/jj#4606]
**Is your feature request related to a problem? Please describe.**

I'd like to be able to configure some git config values with more flexibility, or fallbacks. Ex:

```
[git]
push = 'perso | origin | upstream'
fetch = 'all:matching(perso, origin, upstream)'
```

And I'd need for this to works even if some of the remote does not exists.

It looks like most of the revset possibilities are restricted to specific config keys. It is possible to expand this to other configs?

**Describe the solution you'd like**

Being able to write most config values with a gramar similar to the revset one.

**Describe alternatives you've considered**

**Additional context**

**Comment by yuja** [comment]
Just to be sure, `git.push`/`fetch` don't specify revisions but remote names, so they can't be revsets.

Perhaps, `git.fetch` can be extended to string patterns. Currently it only supports string or a list of strings. `git.push` could be the same, but we'll need to add support for multiple remotes first.
https://martinvonz.github.io/jj/latest/revsets/#string-patterns

**Comment by stacyharper** [comment]
> Just to be sure, git.push/fetch don't specify revisions but remote names, so they can't be revsets.

Yes, I meant something similar to the grammar language we can use for revsets arguments.

> Perhaps, git.fetch can be extended to string patterns. Currently it only supports string or a list of strings. git.push could be the same, but we'll need to add support for multiple remotes first.
> https://martinvonz.github.io/jj/latest/revsets/#string-patterns

You means `regex:"perso|origin|upstream"` could do what I need?

**Comment by yuja** [comment]
> You means `regex:"perso|origin|upstream"` could do what I need?

Yes. There's an idea to add logical operators to string patterns, but that will be a bigger change.

**Comment by PhilipMetzger** [comment]
Only semi-related but I think #3262 also needs a mention here, since some parts of the conversation there could also belong here.
